worker/machiner/machiner.go

```diff
@@ -125,16 +125,30 @@ func setMachineAddresses(tag names.MachineTag, m Machine) error {
 	}
 	var hostAddresses corenetwork.ProviderAddresses
 	for _, addr := range addrs {
-		var ip net.IP
+		var (
+			ip      net.IP
+			netmask net.IPMask
+		)
+
 		switch addr := addr.(type) {
 		case *net.IPAddr:
 			ip = addr.IP
+			netmask = ip.DefaultMask() // may return nil if not ipv4 address
```

Comment:
I might be trying to be too smart here but I think they will typically be /24 so this should work as expected.